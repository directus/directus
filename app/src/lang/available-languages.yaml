# To be sorted by names of languages (not codes)
af-ZA: Afrikaans (South Africa)
sq-AL: Albanian (Albania)
ar-SA: Arabic (Saudi Arabia)
bn-IN: Bengali (India)
bs-BA: Bosnian (Bosnia and Herzegovina)
br-FR: Breton (France)
bg-BG: Bulgarian (Bulgaria)
ca-ES: Catalan (Spain)
zh-CN: Chinese (Simplified)
zh-TW: Chinese (Traditional)
hr-HR: Croatian (Croatia)
cs-CZ: Czech (Czech Republic)
da-DK: Danish (Denmark)
nl-NL: Dutch (Netherlands)
en-CA: English (Canada)
en-GB: English (United Kingdom)
en-US: English (United States)
eo-UY: Esperanto
et-EE: Estonian (Estonia)
fo-FO: Faroese (Faroe Islands)
fa-IR: Farsi (Iran)
fi-FI: Finnish (Finland)
fr-CA: French (Canada)
fr-FR: French (France)
ka-GE: Georgian (Georgia)
de-DE: German (Germany)
el-GR: Greek (Greece)
he-IL: Hebrew (Israel)
hi-IN: Hindi (India)
hu-HU: Hungarian (Hungary)
is-IS: Icelandic (Iceland)
id-ID: Indonesian (Indonesia)
it-IT: Italian (Italy)
ja-JP: Japanese (Japan)
km-KH: Khmer (Cambodia)
ko-KR: Korean (Korea)
kmr-TR: Kurdish Kurmanji (Turkey)
ckb-IR: Kurdish Sorani (Iraq)
lt-LT: Lithuanian (Lithuania)
ms-MY: Malay (Malaysia)
mr-IN: Marathi (India)
mn-MN: Mongolian (Mongolia)
no-NO: Norwegian (Norway)
pl-PL: Polish (Poland)
pt-BR: Portuguese (Brazil)
pt-PT: Portuguese (Portugal)
ro-RO: Romanian (Romania)
ru-RU: Russian (Russian Federation)
sr-SP: Serbian (Cyrillic) (Serbia and Montenegro)
sr-CS: Serbian (Latin) (Serbia and Montenegro)
si-LK: Sinhala (Sri Lanka)
sk-SK: Slovak (Slovakia)
sl-SI: Slovenian (Slovenia)
es-CL: Spanish (Chile)
es-419: Spanish (Latin America)
es-MX: Spanish (Mexico)
es-ES: Spanish (Spain)
sw-KE: Swahili (Kenya)
sw-TZ: Swahili (Tanzania)
sv-SE: Swedish (Sweden)
tg-TJ: Tajik
th-TH: Thai (Thailand)
tr-TR: Turkish (Turkey)
uk-UA: Ukrainian (Ukraine)
vi-VN: Vietnamese (Vietnam)
