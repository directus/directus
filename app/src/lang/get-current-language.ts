import { RTL_LANGUAGES } from '@/constants/language-direction';
import { useServerStore } from '@/stores/server';
import { useUserStore } from '@/stores/user';
import { isIn } from '@directus/utils';
import { get } from 'lodash';

export function getCurrentLanguage(fallback = 'en-US') {
	const usersStore = useUserStore();
	const serverStore = useServerStore();

	const lang = get(
		usersStore,
		['currentUser', 'language'],
		get(serverStore, ['info', 'project', 'default_language'], fallback),
	) ?? fallback;

	let dir = get(usersStore, ['currentUser', 'language_direction'], 'auto');

	if (dir !== 'ltr' && dir !== 'rtl') {
		dir = isIn(lang, RTL_LANGUAGES) ? 'rtl' : 'ltr';
	}

	return { lang, dir };
}
