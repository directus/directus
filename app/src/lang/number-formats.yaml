af-ZA:
  currency:
    style: 'currency'
    currency: 'ZAR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sq-AL:
  currency:
    style: 'currency'
    currency: 'ALL'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ar-SA:
  currency:
    style: 'currency'
    currency: 'SAR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

bn-IN:
  currency:
    style: 'currency'
    currency: 'INR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

bs-BA:
  currency:
    style: 'currency'
    currency: 'BAM'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

br-FR:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

bg-BG:
  currency:
    style: 'currency'
    currency: 'BGN'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ca-ES:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

zh-CN:
  currency:
    style: 'currency'
    currency: 'CNY'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

zh-TW:
  currency:
    style: 'currency'
    currency: 'TWD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

hr-HR:
  currency:
    style: 'currency'
    currency: 'HRK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

cs-CZ:
  currency:
    style: 'currency'
    currency: 'CZK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

da-DK:
  currency:
    style: 'currency'
    currency: 'DKK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

nl-NL:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

en-CA:
  currency:
    style: 'currency'
    currency: 'CAD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

en-GB:
  currency:
    style: 'currency'
    currency: 'GBP'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

en-US:
  currency:
    style: 'currency'
    currency: 'USD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

eo-UY:
  currency:
    style: 'currency'
    currency: 'UYU'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

et-EE:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

fo-FO:
  currency:
    style: 'currency'
    currency: 'DKK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

fa-IR:
  currency:
    style: 'currency'
    currency: 'IRR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

fi-FI:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

fr-CA:
  currency:
    style: 'currency'
    currency: 'CAD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

fr-FR:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ka-GE:
  currency:
    style: 'currency'
    currency: 'GEL'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

de-DE:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

el-GR:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

he-IL:
  currency:
    style: 'currency'
    currency: 'ILS'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

hi-IN:
  currency:
    style: 'currency'
    currency: 'INR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

hu-HU:
  currency:
    style: 'currency'
    currency: 'HUF'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

is-IS:
  currency:
    style: 'currency'
    currency: 'ISK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

id-ID:
  currency:
    style: 'currency'
    currency: 'IDR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

it-IT:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ja-JP:
  currency:
    style: 'currency'
    currency: 'JPY'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

km-KH:
  currency:
    style: 'currency'
    currency: 'KHR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ko-KR:
  currency:
    style: 'currency'
    currency: 'KRW'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

kmr-TR:
  currency:
    style: 'currency'
    currency: 'TRY'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ckb-IR:
  currency:
    style: 'currency'
    currency: 'IRR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

lt-LT:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ms-MY:
  currency:
    style: 'currency'
    currency: 'MYR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

mn-MN:
  currency:
    style: 'currency'
    currency: 'MNT'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

mr-IN:
  currency:
    style: 'currency'
    currency: 'INR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

no-NO:
  currency:
    style: 'currency'
    currency: 'NOK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false
pl-PL:
  currency:
    style: 'currency'
    currency: 'PLN'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

pt-BR:
  currency:
    style: 'currency'
    currency: 'BRL'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

pt-PT:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ro-RO:
  currency:
    style: 'currency'
    currency: 'RON'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

ru-RU:
  currency:
    style: 'currency'
    currency: 'RUB'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sr-SP:
  currency:
    style: 'currency'
    currency: 'RSD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sr-CS:
  currency:
    style: 'currency'
    currency: 'RSD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

si-LK:
  currency:
    style: 'currency'
    currency: 'LKR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sk-SK:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sl-SI:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

es-CL:
  currency:
    style: 'currency'
    currency: 'CLP'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

es-419:
  currency:
    style: 'currency'
    currency: 'USD'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

es-MX:
  currency:
    style: 'currency'
    currency: 'MXN'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

es-ES:
  currency:
    style: 'currency'
    currency: 'EUR'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sw-KE:
  currency:
    style: 'currency'
    currency: 'KES'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sw-TZ:
  currency:
    style: 'currency'
    currency: 'TZS'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

sv-SE:
  currency:
    style: 'currency'
    currency: 'SEK'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

tg-TJ:
  currency:
    style: 'currency'
    currency: 'TJS'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

th-TH:
  currency:
    style: 'currency'
    currency: 'THB'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

tr-TR:
  currency:
    style: 'currency'
    currency: 'TRY'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

uk-UA:
  currency:
    style: 'currency'
    currency: 'UAH'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false

vi-VN:
  currency:
    style: 'currency'
    currency: 'VND'
    notation: standard
  decimal:
    style: 'decimal'
    minimumFractionDigits: 2
    maximumFractionDigits: 2
  percent:
    style: 'percent'
    useGrouping: false
