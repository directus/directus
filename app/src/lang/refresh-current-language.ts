import { useFieldsStore } from '@/stores/fields';
import { useTranslationsStore } from '@/stores/translations';
import { useUserStore } from '@/stores/user';

export async function refreshCurrentLanguage() {
	const fieldsStore = useFieldsStore();
	const userStore = useUserStore();
	const translationsStore = useTranslationsStore();

	try {
		await translationsStore.loadTranslations(userStore.language);

		// Refetch fields in order to translate
		await fieldsStore.hydrate();
	} catch {
		// eslint-disable-next-line no-console
		console.error('Failed loading translations');
	}
}
