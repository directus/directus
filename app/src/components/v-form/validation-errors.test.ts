import type { GlobalMountOptions } from '@/__utils__/types';
import { i18n } from '@/lang';
import type { Field, ValidationError } from '@directus/types';
import { mount } from '@vue/test-utils';
import { expect, test, describe, it } from 'vitest';
import ValidationErrors from './validation-errors.vue';

const global: GlobalMountOptions = {
	plugins: [i18n],
};

test('Mount component', () => {
	expect(ValidationErrors).toBeTruthy();

	const wrapper = mount(ValidationErrors, {
		props: {
			validationErrors: [],
			fields: [],
		},
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

describe('Custom validation message', () => {
	const baseField: Field = {
		collection: 'posts',
		name: 'Title',
		field: 'title',
		type: 'string',
		schema: {
			name: 'title',
			table: 'posts',
			data_type: 'varchar',
			default_value: null,
			max_length: 255,
			numeric_precision: null,
			numeric_scale: null,
			is_generated: false,
			generation_expression: null,
			is_nullable: true,
			is_unique: false,
			is_indexed: false,
			is_primary_key: false,
			has_auto_increment: false,
			foreign_key_column: null,
			foreign_key_table: null,
		},
		meta: {
			id: 1,
			collection: 'posts',
			field: 'title',
			special: null,
			interface: 'input',
			options: null,
			display: null,
			display_options: null,
			readonly: false,
			hidden: false,
			sort: 1,
			width: 'full',
			translations: null,
			note: null,
			conditions: null,
			required: false,
			group: null,
			validation: null,
			validation_message: null,
		},
	};

	const customValidationRule = { _and: [{ title: { _contains: 'a' } }] };

	const customValidationError = {
		field: 'title',
		path: [],
		type: 'contains',
		substring: 'a',
		hidden: false,
		group: null,
	} as unknown as ValidationError;

	const requiredValidationError = {
		field: 'title',
		path: [],
		type: 'nnull',
		hidden: false,
		group: null,
	} as unknown as ValidationError;

	it('appears when custom validation rule fails', () => {
		const wrapper = mount(ValidationErrors, {
			props: {
				validationErrors: [customValidationError],
				fields: [
					{
						...baseField,
						meta: {
							...baseField.meta,
							validation: customValidationRule,
							validation_message: 'my custom message',
						},
					} as Field,
				],
			},
			global,
		});

		expect(wrapper.html()).toContain('my custom message');
	});

	it('appears when required rule fails, but no custom validation rule exists', () => {
		const wrapper = mount(ValidationErrors, {
			props: {
				validationErrors: [requiredValidationError],
				fields: [
					{
						...baseField,
						meta: {
							...baseField.meta,
							validation: null,
							validation_message: 'my custom message',
							required: true,
						},
					} as Field,
				],
			},
			global,
		});

		expect(wrapper.html()).toContain('my custom message');
	});

	it('does not appear when required rule fails and custom validation rule exists', () => {
		const wrapper = mount(ValidationErrors, {
			props: {
				validationErrors: [requiredValidationError],
				fields: [
					{
						...baseField,
						meta: {
							...baseField.meta,
							validation: customValidationRule,
							validation_message: 'my custom message',
							required: true,
						},
					} as Field,
				],
			},
			global,
		});

		expect(wrapper.html()).not.toContain('my custom message');
	});

	it('appears when required rule and custom validation rule fails', () => {
		const wrapper = mount(ValidationErrors, {
			props: {
				validationErrors: [customValidationError, requiredValidationError],
				fields: [
					{
						...baseField,
						meta: {
							...baseField.meta,
							validation: customValidationRule,
							validation_message: 'my custom message',
							required: true,
						},
					} as Field,
				],
			},
			global,
		});

		expect(wrapper.html()).toContain('my custom message');
		expect(wrapper.html()).toContain('Value is required');
	});
});
