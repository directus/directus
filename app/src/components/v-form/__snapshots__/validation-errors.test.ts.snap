// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<v-notice data-v-4508db34="" type="danger" class="full">
  <div data-v-4508db34="">
    <p data-v-4508db34="">The following fields have invalid values:</p>
    <ul data-v-4508db34="" class="validation-errors-list"></ul>
  </div>
</v-notice>"
`;
